src/statsmodels/formula.jl

```diff
@@ -321,8 +323,8 @@ function setcontrasts!(mf::ModelFrame, new_contrasts::Dict)
 end
 setcontrasts!(mf::ModelFrame; kwargs...) = setcontrasts!(mf, Dict(kwargs))
 
-asmatrix(a::AbstractMatrix) = a
-asmatrix(v::AbstractVector) = reshape(v, (length(v), 1))
+asmatrix(T::Type, a::AbstractMatrix) = convert(T, a)
```

Comment:
Since this function is only used in `modelmat_cols`, probably better remove it altogether. You can call `reshape(a, size(a, 1), size(a, 2)))` instead, which works for both vectors and matrices.